src/modules/vmount/output_rc.cpp

```diff
@@ -49,20 +43,17 @@ using math::constrain;
 namespace vmount
 {
 
-OutputRC::OutputRC(const OutputConfig &output_config)
-	: OutputBase(output_config)
+OutputRC::OutputRC(const Parameters &parameters)
+	: OutputBase(parameters)
 {
 }
 
-int OutputRC::update(const ControlData *control_data)
+void OutputRC::update(const ControlData &control_data)
 {
-	if (control_data) {
```

Comment:
This was to avoid potentially expensive ops when there was no update. Would be good to keep.

Reply:
We're only really saving 7 copies with that, no actual calculations.

Reply:
It's more than that. This is what I mean: https://github.com/PX4/PX4-Autopilot/blob/c5f04afbbfdfae990bc16686bef3ca85bd48b999/src/modules/vmount/output.cpp#L130

Reply:
But that function was outside of this `if`? And if called it's then ignored unless `if (control_data.type != ControlData::Type::LonLat) {`.

Or what am I missing?

Reply:
It's the indirect call from https://github.com/PX4/PX4-Autopilot/blob/c5f04afbbfdfae990bc16686bef3ca85bd48b999/src/modules/vmount/output.cpp#L115 (Yes only in case of `LonLat`)

Reply:
In which case it probably should re-calculate it, so I think that makes sense.

Reply:
No, only if there was an update. This was correct as it was.

Reply:
I think this is now addressed.